Kubernetes Dashboard: update deployment instructions

stefsuse opened the pull request:
Update deployment instructions for Kubernetes dashboard. 
 - install a newer version of the dashboard that includes many UX and security improvements and better Heapster integration. 
 - use helm for deployments
 - NodePort is the preferred method of exposing services

Tested on the latest development build of CaaSP.

stefsuse commented:
@cwickert  pinging you because you have recent commits. Can I get a review for this change?

stefsuse commented:
@cjschroder @lproven @fsundermeyer 
pinging a few more folks from another PR. I am not in the SUSE org so I can't add anyone. Can anyone spare a quick review, or tell me how I am supposed to request one? Thanks.

cjschroder commented:
Hi @stefsuse, requesting reviewers here usually works. I'll have a look.

fsundermeyer commented on xml/deployment_additional.xml:
```diff
@@ -68,36 +68,73 @@ if necessary.
+  <para>
+  On login cluster resources and basic metrics are populated. 
+  </para>
+
```
Maybe add a small paragraph exaplaining what has been done and what needs to be done next? E,g, (no guarantee on the content - I do not even know what this is all about)

&lt;para>
 Now that the &kube; Dashboard has been successfully installed, it needs to be exposed to the worker nodes:
&lt;/para>

fsundermeyer commented on xml/deployment_additional.xml:
```diff
@@ -68,36 +68,73 @@ if necessary.
     <para>
-     In your browser open <emphasis>http://127.0.0.1:8001/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard/#!/workload?namespace=default</emphasis>. The
-     &kube; dashboard is running there.
+      Visit <literal>https://WORKER_NODE_ADDRESS:NODE_PORT</literal> in your browser.
```
Visit &lt;literal>https://&lt;replaceable>WORKER_NODE_ADDRESS&lt;/replaceable>:&lt;replaceable>NODE_PORT&lt;/replaceable>&lt;/literal> in your browser.

* Why visit this address?
* What to do once I have opened the page in the browser?
* Does this have to be done on each worker node?

stefsuse replied:
Is there a document somewhere that explains what replaceable is? I compared the output in my generated PDF and it appeared the same, so I assumed <literal> worked like HTML <pre>

stefsuse replied:
This section is just breaking out exposing the dashboard from the rest of the setup, because not every customer might want that. I can add a clarification that this URL is for access from outside of the cluster though.

fsundermeyer commented on xml/deployment_additional.xml:
```diff
@@ -68,36 +68,73 @@ if necessary.
+
+  <procedure>
+   <title>Exposing the Dashboard</title>
+   <step>
```
&lt;para>
 Run the following command on ??? as user ???
&lt;/para>

cjschroder commented:
I will fix the formatting and XML tags-- @stefsuse please take care of the content corrections. Thanks for a good review @fsundermeyer, and thanks for the  PR, Stef.

stefsuse commented on xml/deployment_additional.xml:
```diff
@@ -68,36 +68,73 @@ if necessary.
+    <para>
+      helm version 2.7.2+ and kubectl version 1.8.0+ recommended
+    </para>
+  </note>
```
I think what the comment is saying is there should not be indentations to the content as well as other changes? Fixed.

stefsuse commented on xml/deployment_additional.xml:
```diff
@@ -68,36 +68,73 @@ if necessary.
     <para>
-     After the previous command finishes, run the following to expose the
-     proxy:
+      Select <literal>token</literal> authentication. To retrieve your token refer to the value in your kubeconfig file by running the command:
```
The kubeconfig is the file the user downloads from Velum after bootstrap. It can be placed anywhere on the host system so I did not want to put in an explicit path.

stefsuse commented:
Addressed comments and left some follow up comments here. Carla, I think I addressed the tags as well but still feel free to make edits if it still does not look right. (the PDF looks good on my desktop..)

cjschroder commented on xml/deployment_additional.xml:
```diff
@@ -62,42 +62,75 @@ if necessary.
     <para>
-     Connect to the &admin_node; using SSH.
+      Visit <literal>http://127.0.0.1:8081/</literal> in your browser. 
+      You will be greeted with by a welcome page containing a dialog to configure authentication.
```
Delete "with"

stefsuse replied:
thanks! I'll look out for your PR

cjschroder commented:
This has become confusing :) I'm going to merge & then see if it needs any tweaks. Thanks everyone!

cwickert commented:
@stefsuse Does this apply to CaaS 2 as well?
@cjschroder Did you make additional changes (that are not part of your PR #19)

cjschroder commented:
@cwickert , no that is all.

stefsuse commented:
@cwickert Yes these instructions apply to V2
